3_3.js

```diff
@@ -0,0 +1,59 @@
+// Создайте калькулятор позволяющий добавлять в него дополнительные методы и сохранять результат, 
+// по умолчанию должны присутствовать методы add, substract
+// Пример: 
+// const calculator = new Calc()
+// calculator.operation('31 + 32') // 63
+// calculator.operation('10 * 2') // 10
+// calculator.addOperation('/', (a, b) => a / b)
+// calculator.operation('10 / 2') // 5
+// Также, он должен хранить историю всех операций и выводить ее по запросу: 
+// calculator.history() /* [{operation: '+', operands: [31,32]}, {operation: '*', 
+// operands: [10,2]}, {operation: '/', operands: [10,2]}] */ 
+// И очищать историю
+// calculator.clearHistory()
+// calculator.history() // []
+// console.log(eval('10 * 2'))
+
+function Calc() {
+  let history = []
+  var i = -1
+
+  this.history = function() {return history};
+  
+  this.clearHistory = function() {history = []}
+      
+  this.operation = function(str) {
+      split = str.split(' '),
+      a = +split[0],
+      b = +split[2],
+      act = split[1]
+      history[++i] = {}
+      history[i]['operation'] = split[1],
+      history[i]['operand'] = [Number(a),Number(b)]
+      return this.action[act](a, b);
```

Comment:
Запись в историю можно упростить используя метод [`Array.push()`](https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/Array/push)
```suggestion
      history.push({operation: act, operand: [a, b]})
      return this.action[act](a, b);
```